packages/downloads/src/portal-poll-export-job-status.ts

```diff
@@ -0,0 +1,233 @@
+        portalPollExportJobStatus(params, eventEmitter, pollingInterval);
+      }, pollingInterval);
+    }).catch((error:any) => {
+      return eventEmitter.emit(`${downloadId}PollingError`, { detail: { error } });
+    });
+}
+
+function completedHandler (params: any) {
+  const {
+    portal,
+    datasetId,
+    format,
+    username,
+    token,
+    downloadId,
+    exportCreated,
+    spatialRefId,
+    spatialRefWkt,
+    eventEmitter
+  } = params;
+
+  const [ itemId ] = datasetId.split('_');
+
+  return new Promise((resolve, reject) => {
+    getExportsFolder({
+      portal,
+      username,
+      token
+    }).then(exportFolderId => {
+      Promise.all([
+        // @ts-ignore
+        updateItem({
+          portal,
+          item: {
+            id: itemId,
+            typeKeywords: `export:${datasetId},modified:${exportCreated},exportFormat:${format},spatialRef:${spatialRefId || spatialRefWkt}`
+          },
+          owner: username,
+          params: { token } 
+        }),
+        // @ts-ignore
+        moveItem({
+          portal,
+          itemId: downloadId,
+          folderId: exportFolderId,
+          owner: username,
+          params: { token } 
+        }),
+        shareItemAsOwner({
+          portal,
+          exportItemId: downloadId,
+          username,
+          token
+        })
+      ])
+    }).then(results => {
+      eventEmitter.emit(`${downloadId}ExportComplete`, { detail: { metadata: {
+        downloadId,
+        status: 'ready',
+        downloadUrl: urlBuilder({ host: portal, route: `content/items/${downloadId}/data`, query: { token }})
+      } } });
+      resolve();
+    }) 
+    .catch(reject);
+  })
+}
+
+function getExportsFolder(params: any): Promise<any> {
+  return new Promise((resolve, reject) => {
+    getUserContent(params)
+      .then(userContent => {
+        const exportFolderId = userContent.folders.find((folder: any) => { return folder.title === 'item-exports'});
+      if (exportFolderId) {
+        resolve(exportFolderId);
+        return { then: () => {} };
+      }
+      return createFolder({ ...params, title: 'item-exports'});
+      }).then(result => {
+        return resolve(result.folder.id);
+      }).catch(reject);
+  })
+}
+
+function getUserContent (params: any): Promise<any> {
+  const {
+    portal,
+    username,
+    token
+  } = params;
+
+  const url = urlBuilder({ host: portal, route: `content/users/${username}`, query: { token, f: 'json' } })
+  return new Promise((resolve, reject) => {
+    fetch(url).then(resp => {
+      const { ok, status, statusText } = resp;
+      if (!ok) {
+        throw new RemoteServerError(statusText, url, status);
+      }
+      return resp.json();
+    }).then(json => {
+      if (json.error) {
+        throw new RemoteServerError(json.error.message, url, json.error.code);
+      }
+      return resolve(json);
+    })
+    .catch(reject);
+  });
+}
+
+function createFolder (params: any): Promise<any> {
```

Comment:
This exists in rest-js --> https://esri.github.io/arcgis-rest-js/api/portal/createFolder/